parsl/tests/test_monitoring/test_file_provenance.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+import os
+import shutil
+import time
+
+import parsl
+from parsl.config import Config
+from parsl.executors import HighThroughputExecutor
+from parsl.monitoring import MonitoringHub
+from parsl.providers import LocalProvider
+from parsl.data_provider.files import File
+
+
+@parsl.bash_app
+def initialize(outputs=[]):
+    import time
+    time.sleep(1)
+    return f"echo 'Initialized' > {outputs[0]}"
+
+
+@parsl.python_app
+def split_data(inputs=None, outputs=None):
+    with open(inputs[0], 'r') as fh:
+        data = fh.read()
+    for i, op in enumerate(outputs):
+        with open(op, 'w') as ofh:
+            ofh.write(f"{i + 1}\n")
+            ofh.write(data)
+
+
+@parsl.python_app
+def process(inputs=None, outputs=None):
+    with open(inputs[0], 'r') as fh:
+        data = fh.read()
+    with open(outputs[0], 'w') as ofh:
+        ofh.write(f"{data} processed")
+
+
+@parsl.python_app
+def combine(inputs=None, outputs=None):
+    with open(outputs[0], 'w') as ofh:
+        for fl in inputs:
+            with open(fl, 'r') as fh:
+                ofh.write(fh.read())
+                ofh.write("\n")
+
+
+def fresh_config(run_dir):
+    return Config(
+        run_dir=str(run_dir),
+        executors=[
+            HighThroughputExecutor(
+                address="127.0.0.1",
+                label="htex_local",
+                provider=LocalProvider(
+                    init_blocks=1,
+                    min_blocks=1,
+                    max_blocks=1,
+                )
+            )
+        ],
+        strategy='simple',
+        strategy_period=0.1,
+        monitoring=MonitoringHub(
+                        hub_address="localhost",
+                        hub_port=55055,
```

Comment:
in PR #3710 I removed a bunch of `hub_port` from the test suite - you could that here too

Reply:
ok